.circleci/config.yml

```diff
@@ -1,16 +1,31 @@
-version: 2
+version: 2.1
+
+defaults: &defaults
+  docker:
+    - image: srclosson/grafana-plugin-ci-alpine:latest
+
 jobs:
   build:
-    docker:
-      - image: srclosson/grafana-plugin-ci-alpine:latest
+    <<: *defaults
     steps:
       - checkout
+      - restore_cache:
+          keys:
+            - go-modules-v1-{{ checksum "go.sum" }}
```

Comment:
I think deriving the cache key from go.mod is the way to go, since go.sum doesn't lock any versions, it only contains checksums:
```suggestion
            - go-modules-v1-{{ checksum "go.mod" }}
```

Reply:
Maybe I'm confused by old documentation? Reading [here](https://blog.golang.org/using-go-modules):
> The go command uses the go.sum file to ensure that future downloads of these modules retrieve the same bits as the first download, to ensure the modules your project depends on do not change unexpectedly, whether for malicious, accidental, or other reasons.

All the articles I've read about CircleCI and go modules cache uses go.sum and their official orb also uses that, see https://github.com/CircleCI-Public/go-orb/blob/master/src/commands/save-cache.yml (circle ci orb registry seems to be down currently so cannot link to that).

Reply:
@marefr You can see [here](https://github.com/golang/go/wiki/Modules#is-gosum-a-lock-file-why-does-gosum-include-information-for-module-versions-i-am-no-longer-using) that go.mod is enough for reproducible builds, go.sum is not a lock file.

The documentation you quote, as I understand it, only confirms that go.sum is just a collection of checksums for verifying package integrity?

Reply:
I have changed, but I'm still not sure what's the right thing.

Reply:
Thanks. I'm also unsure why people are using go.sum, but considering that go.mod is considered the source of truth (for 100% reproducible builds) I don't understand why you would derive the cache key from go.sum.